torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -609,9 +666,14 @@ def _should_free_in_backward(
     Returns whether FSDP should free the unsharded flattened parameter in the
     post-backward or not.
     """
-    return (
-        state._sync_gradients and handle.uses_sharded_strategy
-    ) or handle._config.sharding_strategy == HandleShardingStrategy.FULL_SHARD
+    # We always free if we are syncing gradients (i.e. not in no_sync) and parameters
+    # are sharded.
+    free_unsharded = state._sync_gradients and handle.uses_sharded_strategy
+    # For NO_SHARD we don't need to free full parameters, for Zero-2 strategies, we skip
```

Comment:
nit: Let us stylize following the paper?
```suggestion
    # For NO_SHARD we don't need to free full parameters, for ZeRO-2 strategies, we skip
```